source/gpus_in_openstack.rst

```diff
@@ -4,6 +4,601 @@
+   vgpu_driver_url: "http://{{ pulp_url }}/pulp/content/nvidia/NVIDIA-GRID-Linux-KVM-525.105.14-525.105.17-528.89.zip"
+
+See :ref:`NVIDIA Role Configuration`.
+
+.. _NVIDIA control host:
+
+Placing the GRID driver on the ansible control host
+---------------------------------------------------
+
+Copy the driver bundle to a known location on the ansible control host. Set the ``vgpu_driver_url`` configuration variable to reference this
+path using ``file`` as the url scheme e.g:
+
+.. code:: yaml
+
+    # Location of NVIDIA GRID driver on localhost
+    vgpu_driver_url: "file://{{ lookup('env', 'HOME') }}/NVIDIA-GRID-Linux-KVM-525.105.14-525.105.17-528.89.zip"
+
+See :ref:`NVIDIA Role Configuration`.
+
+.. _NVIDIA OS Configuration:
+
+OS Configuration
+----------------
+
+Host OS configuration is done by using roles in the `stackhpc.linux <https://github.com/stackhpc/ansible-collection-linux>`_ ansible collection.
+
+Add the following to your ansible ``requirements.yml``:
+
+.. code-block:: yaml
+   :caption: $KAYOBE_CONFIG_PATH/ansible/requirements.yml
+
+    #FIXME: Update to known release When VGPU and IOMMU roles have landed
+    collections:
+      - name: stackhpc.linux
+        source: git+https://github.com/stackhpc/ansible-collection-linux.git,preemptive/vgpu-iommu
+        type: git
+
+Create a new playbook or update an existing on to apply the roles:
+
+.. code-block:: yaml
+   :caption: $KAYOBE_CONFIG_PATH/ansible/host-configure.yml
+
+    ---
+
+      - hosts: iommu
+        tags:
+          - iommu
+        tasks:
+          - import_role:
+              name: stackhpc.linux.iommu
+        handlers:
+          - name: reboot
+            set_fact:
+              kayobe_needs_reboot: true
+
+      - hosts: vgpu
+        tags:
+          - vgpu
+        tasks:
+          - import_role:
+              name: stackhpc.linux.vgpu
+        handlers:
+          - name: reboot
+            set_fact:
+              kayobe_needs_reboot: true
+
+      - name: Reboot when required
+        hosts: overcloud
+        vars:
+          # This playbook will execute after nodes are deployed
+          # and before overcloud host configure - we can't assume
+          # users and venvs exist.
+          ansible_user: "cloud-user"
+          ansible_python_interpreter: "/usr/bin/python3"
+        tags:
+          - reboot
+        tasks:
+          - name: Reboot
+            reboot:
+              reboot_timeout: 3600
+            become: true
+            when: kayobe_needs_reboot | default(false) | bool
+
+.. _NVIDIA Role Configuration:
+
+Role configuration
+^^^^^^^^^^^^^^^^^^
+
+Configure the location of the NVIDIA driver:
+
+.. code-block:: yaml
+   :caption: $KAYOBE_CONFIG_PATH/vgpu.yml
+
+    ---
+
+    vgpu_driver_url: "http://{{ pulp_url }}/pulp/content/nvidia/NVIDIA-GRID-Linux-KVM-525.105.14-525.105.17-528.89.zip"
+
+Configure the VGPU devices:
+
+.. code-block:: yaml
+   :caption: $KAYOBE_CONFIG_PATH/ansible/group_vars/compute_vgpu/vgpu
```

Comment:
Should this go in the inventory instead?

Reply:
Updated the caption to reflect this